File: Biblioteca/funcoes.py
from tabulate import tabulate
# import emoji

def adicionar_livro(listaLivros):
    titulo = input("Digite o titulo do livro: ")
    autor = input("Digite o nome do autor do livro: ")
    
    for livro in listaLivros:
        if livro["Título"].lower().strip() == titulo.lower().strip() and livro["Autor"].lower().strip() == autor.lower().strip():
            print("Esse livro já foi cadastrado")
            return

    livro = {
        "Título": titulo, 
        "Autor": autor,
        "Status": "Disponível" 
    }
    
    listaLivros.append(livro)
    print("Livro cadastrado com sucesso \u2705")
    return

def emprestar_livro(listaLivros):
    titulo = input("Digite o título do livro que deseja retirar: ").lower()
    for livro in listaLivros:
        if titulo.lower().strip() == livro["Título"].lower().strip():
            if livro["Status"] == "Disponível":
                livro["Status"] = "Emprestado"
                print("Livro retirado com sucesso")
                return
            if livro["Status"] == "Emprestado":
                print("Livro não disponível no momento")
                return
        if livro not in listaLivros: 
            print("\U0001f6a8 Livro não cadastrado")
            return

def devolver_livro(listaLivros):
    titulo = input("Digite o título do livro que deseja devolver: ")
    for livro in listaLivros:
        if titulo.lower().strip() == livro["Título"].lower().strip():
            if livro["Status"] == "Emprestado":
                livro["Status"] = " Disponível"
                print("Livro devolvido com sucesso \u2705")
                return
            if livro["Status"] == "Disponível":
                print("O livro não foi emprestado, logo não pode ser devolvido")
                return
        if livro not in listaLivros: 
            print("\U0001f6a8 Livro não cadastrado")
            return

def exibir_livros(listaLivros):
    if not listaLivros:
        print("Nenhum livro cadastrado")
        return
    tabela = [[livro["Título"], livro["Autor"], livro["Status"]] for livro in listaLivros]
    print(tabulate(tabela, headers=["Título", "Autor", "Status"], tablefmt="fancy_grid"))

File: BibliotecaJs/funcoes.js
function adicionar_livro(listaLivros) {
    let titulo = prompt("Digite o título do livro");
    let autor = prompt("Digite o autor do livro");

    for (let livro in listaLivros) {
        if (titulo == livro["Título"] && autor == livro["Autor"]) {
            return alert("Livro já cadastrado")
        };
    };

    let livro = {
        Título: titulo, 
        Autor: autor,
        Status: "Disponível"
    };

    listaLivros.push(livro);
    return alert("Livro cadastrado com sucesso");
};

function emprestar_livro(listaLivros) {
    let titulo = prompt("Digite o livro que deseja retirar")

    for (let livro in listaLivros) {
        if (titulo == livro["Título"]) {
            if (livro("Status") == "Disponível") {
                livro["Status"] = "Emprestado"
                return alert("Livro retirado com sucesso");
            };

            if (livro["Status"] == "Emprestado") {
                return alert("Livro não disponível no momento");
            };
        };

        if (!listaLivros.includes(livro)) {
           return alert("Livro não cadastrado")
        };
    };
};

function devolver_livro(listaLivros) {
    let titulo = prompt("Digite o nome do livro que deseja devolver");

    for (let livro in listaLivros) {
        if (titulo == livro["Título"]) {
            if (livro["Status"] == "Emprestado") {
                livro["Status"] = "Disponível"
                return alert("Livro devolvido com sucesso")
            };

            if (livro["Status"] == "Disponível") {
                return alert("O livro não foi emprestado, logo não pode ser devolvido")
            };
        };

        if (!listaLivros.includes(livro)) {
           return alert("Livro não cadastrado")
        };
    };
};

function exibir_livros(listaLivros) {
    if (listaLivros.length === 0) {
        return alert("Nenhum livro cadastrado");
    };
};
